### PR TITLE
[CALCITE] Add support for DEALLOCATE PREPARE call in CREATE PROCEDURE statement

### PR DESCRIPTION
**Pull request by michaelvdow**

### CODE REVIEW

File: core/src/main/java/org/apache/calcite/sql/SqlDeallocatePrepare.java [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+  private static final SqlSpecialOperator OPERATOR =
+      new SqlSpecialOperator("DEALLOCATE PREPARE", SqlKind.DEALLOCATE_PREPARE);
+
+  public final SqlIdentifier statementName;
+
+  /**
+   * Creates a {@code SqlDeallocatePrepare}.
+   *
+   * @param pos  Parser position, must not be null
+   * @param statementName  Name of the statement to deallocate, must not be null
```
Comment (dasch-1) [inline core/src/main/java/org/apache/calcite/sql/SqlDeallocatePrepare.java:38]: nit: Line wrap

File: parsing/dialects/intermediate/dialects/dialect1/config.fmpp [hunk 2]
```diff
@@ -77,6 +77,7 @@ data: {
       "org.apache.calcite.sql.SqlCreateTableDialect1",
       "org.apache.calcite.sql.SqlDateTimeAtLocal",
       "org.apache.calcite.sql.SqlDateTimeAtTimeZone",
+      "org.apache.calcite.sql.SqlDeallocatePrepare",
```
Comment (dasch-1) [inline parsing/dialects/intermediate/dialects/dialect1/config.fmpp:80]: nit: Put in alphabetical order (e.g. after SqlDdlNodes). Also, can you remove the duplicate SqlDdlNodes while you're here?

Comment (michaelvdow) [reply parsing/dialects/intermediate/dialects/dialect1/config.fmpp:80]: Yeah I'll update that

File: core/src/main/java/org/apache/calcite/sql/SqlDeallocatePrepare.java (hunk 1 above)
Comment (michaelvdow) [reply core/src/main/java/org/apache/calcite/sql/SqlDeallocatePrepare.java:38]: It looks like this line is also just at 80 chars. Is it supposed to be < 80 or is <=80 fine?

Comment (dasch-1) [reply core/src/main/java/org/apache/calcite/sql/SqlDeallocatePrepare.java:39]: You're right - again! I just realized the helper utility I'm using to determine line lengths in these PRs had an off-by-one error. Should be fixed now.

**Pull request merged**